ui/vic-ui-service/src/main/java/com/vmware/vicui/VicUIServiceImpl.java

```diff
@@ -57,35 +61,30 @@ private static VimPortType initializeVimPort() {
 	}
 
 	static {
-	      HostnameVerifier hostNameVerifier = new HostnameVerifier() {
-	         @Override
-	         public boolean verify(String urlHostName, SSLSession session) {
-	            return true;
-	         }
-	      };
-	      HttpsURLConnection.setDefaultHostnameVerifier(hostNameVerifier);
-
-	      javax.net.ssl.TrustManager[] trustAllCerts = new javax.net.ssl.TrustManager[1];
-	      javax.net.ssl.TrustManager tm = new TrustAllTrustManager();
-	      trustAllCerts[0] = tm;
-	      javax.net.ssl.SSLContext sc = null;
-
-	      try {
-	         sc = javax.net.ssl.SSLContext.getInstance("TLSv1.2");
-	      } catch (NoSuchAlgorithmException e) {
-	         _logger.info(e);
-	      }
-
-	      javax.net.ssl.SSLSessionContext sslsc = sc.getServerSessionContext();
-	      sslsc.setSessionTimeout(0);
-	      try {
-	         sc.init(null, trustAllCerts, null);
-	      } catch (KeyManagementException e) {
-	         _logger.info(e);
-	      }
-	      javax.net.ssl.HttpsURLConnection.setDefaultSSLSocketFactory(
-	            sc.getSocketFactory());
-	   }
+		HostnameVerifier hostNameVerifier = new ThumbprintHostNameVerifier();
+		HttpsURLConnection.setDefaultHostnameVerifier(hostNameVerifier);
+
+		javax.net.ssl.TrustManager[] tms = new javax.net.ssl.TrustManager[1];
+		javax.net.ssl.TrustManager tm = new ThumbprintTrustManager();
+		tms[0] = tm;
+		javax.net.ssl.SSLContext sc = null;
+
+		try {
+			sc = javax.net.ssl.SSLContext.getInstance("SSL");
+		} catch (NoSuchAlgorithmException e) {
+			_logger.info(e);
```

Comment:
@hmahmood I think it's ok to continue here unless you see something else that concerns you. It worked pretty solid for me.